sygnal/apnspushkin.py

```diff
@@ -213,6 +213,9 @@ async def make_apns() -> aioapns.APNs:
         # without this, aioapns will retry every second forever.
         self.apns_client.pool.max_connection_attempts = 3
 
+        # without this, aioapns will not use the proxy if one is configured.
+        self.apns_client.pool.loop = loop
```

Comment:
I'm slightly hesitant to gut wrench into the depths of classes TBH, though appreciate this might be the best option.

Given this proxy seems to be a global value, I wonder if we could create the new loop at startup and do `asyncio.set_event_loop(loop)`?

Reply:
I tried doing that, in many many different ways but couldn't get aioapns to actually behave and use that loop.
So I gave up and resorted to this method (which I also don't like)

Maybe there is a way to do it with setting the event loop, but I wasn't able to make it work.

Reply:
Huh, wtf. Having looked at the code it seems that it just does `asyncio.get_event_loop()` in the constructor, but maybe something is somehow constructing a class at import time or something? Weird.